### DISCUSSION TITLE
magick doesn't work while convert still does

### DISCUSSION THREAD

**Discussion by ArthurPeabody** [ImageMagick/ImageMagick#7400] [Help] [answered]
### ImageMagick version

7.1.1-33

### Operating system, version and so on

 Linux, kernel 6.9.4; Slackware, latest

### Description

magick -append A60377.jpg Caption.jpg out.jpg 

yields: 

magick: no images found for operation `-append' at CLI arg 1 @ error/operation.c/CLIOption/5479.

 but the same command with convert works, except I also get the message that convert has been deprecated.

**Comment by snibgo** [comment, marked as answer]
> no images found for operation `-append'

`magick` will do operations in the order you give them. You are asking to `-append` images *before* you have read them. So move `-append` to *after* reading them.

**Reply by ArthurPeabody** [reply]
Thanks.  convert didn't mind.

**Reply by snibgo** [reply]
Yes. `convert` would re-arrange the execution order in ways that were not easily predictable. `magick` doesn't rearrange the order.
